### PR TITLE
[GCP] Add GCP Compute Metricset

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds GCP Compute Metricset

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [X] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues

- Relates #490

### CODE REVIEW

**Comment by legoguy1000** [comment]
@endorama next one ready for your review

**Comment by legoguy1000** [comment]
@endorama Any thoughts on this one?

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me! Waiting for @endorama 's review 🙂

File: packages/gcp/data_stream/compute/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+metricsets: ["compute"]
+period: {{period}}
+project_id: {{project_id}}
+{{#if credentials_file}}
+credentials_file: {{credentials_file}}
```
Comment (endorama) [inline packages/gcp/data_stream/compute/agent/stream/stream.yml.hbs:5]:
```suggestion
credentials_file_path: {{credentials_file}}
```

[While testing](https://github.com/elastic/integrations/pull/2312/commits/cb31f893b8da21af526fd0c14ad9624ed60aec18) the `billing` package I discovered the key should be `credentials_file_path`.

File: packages/gcp/data_stream/compute/fields/agent.yml [hunk 2]
```diff
@@ -0,0 +1,198 @@
+- name: cloud
+  title: Cloud
+  group: 2
+  description: Fields related to the cloud or infrastructure the events are coming from.
+  footnote: 'Examples: If Metricbeat is running on an EC2 host and fetches data from its host, the cloud info contains the data about this machine. If Metricbeat runs on a remote machine outside the cloud and fetches data from a service running in the cloud, the field contains cloud data from the machine the service is running on.'
```
Comment (endorama) [inline packages/gcp/data_stream/compute/fields/agent.yml:5]:
```suggestion
  footnote: 'Examples: If Metricbeat is running on an GCP Compute VM and fetches data from its host, the cloud info contains the data about this machine. If Metricbeat runs on a remote machine outside the cloud and fetches data from a service running in the cloud, the field contains cloud data from the machine the service is running on.'
```

File: packages/gcp/docs/compute.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+| data_stream.type | Data stream type. | constant_keyword |
+| ecs.version | ECS version this event conforms to. `ecs.version` is a required field and must exist in all events. When querying across multiple indices -- which may conform to slightly different ECS versions -- this field lets integrations adjust to the schema version of the events. | keyword |
+| error | These fields can represent errors of any kind. Use them for errors that happen while fetching events or in cases where the event itself contains an error. | group |
+| error.message | Error message. | match_only_text |
+| event.dataset | Event dataset | constant_keyword |
+| event.module | Event module | constant_keyword |
+| gcp.compute.instance.cpu.reserved_cores.value | Number of cores reserved on the host of the instance | double |
+| gcp.compute.instance.cpu.usage_time.value | Usage for all cores in seconds | double |
+| gcp.compute.instance.cpu.utilization.value | The fraction of the allocated CPU that is currently in use on the instance | double |
+| gcp.compute.instance.disk.read_bytes_count.value | Count of bytes read from disk | long |
```
Comment (endorama) [inline packages/gcp/docs/compute.md:110]: I just noticed that most of these fields are different from the values in gcp metricset. For example: https://github.com/elastic/beats/blob/e6e65aa92fe355c95789691ebf5a3bcecaf5b4ea/x-pack/metricbeat/module/gcp/compute/_meta/fields.yml#L21

I'm wondering if this is ok or beats and integrations are expected to have the same fields (I would expect the latter). Pinging @mtojek here.

@legoguy1000 which version of beats you started this migration from? It seems has old fields values.

**Review by endorama — CHANGES_REQUESTED** [review]

**Comment by endorama** [comment]
Sorry for the delay, I've been stuck adding tests to the `billing` data stream. I did a first round of review and noticed something strange with field names.

File: packages/gcp/docs/compute.md (hunk 3 above)
Comment (legoguy1000) [reply packages/gcp/docs/compute.md:110]: Its totally possible I copied the fields from an old branch. I'll go to master and pull the latest.

**Comment by legoguy1000** [comment]
@endorama updates pushed

**Comment by endorama** [comment]
@legoguy1000 May you please check dashboards too? It seems those still contains old values.

**Review by endorama — COMMENTED** [review]
Thank you for the updates! I see tests are still failing, but wasn't able to pin point the exact issue.

**Comment by endorama** [comment]
rerunning tests in case has been a transactional issue.

/test

**Comment by legoguy1000** [comment]
@endorama It looks to be a mismatch between the sample _event.json and the defined fields for the compute dataset.  The sample events from Metricbeat, https://github.com/elastic/beats/tree/master/x-pack/metricbeat/module/gcp/compute/_meta, were not updated for the new field names and as such don't match the fields.yml.  Do you happen to have updated sample events that could be used?

**Comment by endorama** [comment]
@legoguy1000 I'm working on updating the sample data, thanks for the hint

**Comment by endorama** [comment]
I've been able to update the sample data and working on a PR in beats to update them.

**Comment by legoguy1000** [comment]
> I've been able to update the sample data and working on a PR in beats to update them.

Cool, feel free to push that update to this branch or send it to me and I'll update it.

**Comment by endorama** [comment]
@legoguy1000 I started working on updating the sample events in beats in the linked PR, but is taking longer than expected. In any case I've been able to update the `compute` metricset sample events, so you may take those to update this PR, and I expect that to work. Meanwhile I'll work on updating the beats side too.

**Comment by legoguy1000** [comment]
@endorama looks like it worked.

**Comment by endorama** [comment]
Thank you! I'm merging this as is good to go, I'll open a separate PR with testing for this data stream.

@legoguy1000 next week I'll be off, so feel free to open the next PR but I won't be able to look at it before the 10th of January. Thank you for the contribution and happy new year!

**Pull request merged**